MAVProxy/modules/mavproxy_link.py

```diff
@@ -749,19 +755,26 @@ def accumulated_statustext(self):
 
     def mavlink_packet(self, msg):
         '''handle an incoming mavlink packet'''
-        type = msg.get_type()
-
-        if type == 'HEARTBEAT' or type == 'HIGH_LATENCY2':
-            sysid = msg.get_srcSystem()
-            if not sysid in self.vehicle_list and msg.type != mavutil.mavlink.MAV_TYPE_GCS:
-                self.vehicle_list.add(sysid)
+        pass
 
     def master_callback(self, m, master):
         '''process mavlink message m on master, sending any messages to recipients'''
-
-        # see if it is handled by a specialised sysid connection
         sysid = m.get_srcSystem()
+        compid = m.get_srcComponent()
         mtype = m.get_type()
+
+        if mtype == 'HEARTBEAT' or mtype == 'HIGH_LATENCY2':
+            if not sysid in self.vehicle_list and m.type != mavutil.mavlink.MAV_TYPE_GCS:
+                self.vehicle_list.add(sysid)
+            if not (sysid, compid) in self.mpstate.vehicle_link_map[master.linknum] and m.type != mavutil.mavlink.MAV_TYPE_GCS:
+                self.mpstate.vehicle_link_map[master.linknum].append((sysid, compid))
+                # Newly detected vehicle/link combination, initialise the stream
+                master.mav.request_data_stream_send(sysid, compid,
```

Comment:
There's no equivalent of this in the old code?

I think we should leave this for the scheduled stream-rate-setting bit to avoid having two places doing this.

Reply:
Sure - will remove